### PR TITLE
worked on update user password changes

### PR DESCRIPTION
**Pull request by Surendhar1997**
when old password was wrong and when new password matches with old. Also made tweaks for hierarchy level password changes

### CODE REVIEW

File: security/src/main/java/com/fincity/security/service/UserService.java [hunk 1]
```diff
@@ -652,54 +653,62 @@ public Mono<Boolean> updateNewPassword(String urlAppCode, String urlClientCode,
...
 	}
 
 	public Mono<Boolean> checkPasswordEquality(User u, RequestUpdatePassword reqPassword) {
 
 		if (u.isPasswordHashed())
 			return Mono.just(passwordEncoder.matches(u.getId() + reqPassword.getOldPassword(), u.getPassword())
-			        && passwordEncoder.matches(u.getId() + reqPassword.getNewPassword(), u.getPassword()));
+			        && !passwordEncoder.matches(u.getId() + reqPassword.getNewPassword(), u.getPassword()));
 
 		return Mono.just(StringUtil.safeEquals(reqPassword.getNewPassword(), u.getPassword())
```
Comment (serialcoder13) [inline security/src/main/java/com/fincity/security/service/UserService.java:713]: Need to add the ! here in this line above.

**Review by serialcoder13 — CHANGES_REQUESTED** [review]
Make the changes.

File: security/src/main/java/com/fincity/security/service/UserService.java (hunk 1 above)
Comment (Surendhar1997) [reply security/src/main/java/com/fincity/security/service/UserService.java]: updated the changes

**Review by serialcoder13 — APPROVED** [review]
LGTM

**Pull request merged**